sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -445,7 +445,7 @@ OutputT applyTransform(String name, InputT input,
   private Set<String> usedFullNames = new HashSet<>();
   private CoderRegistry coderRegistry;
   private final List<String> unstableNames = new ArrayList<>();
-  private final PipelineOptions defaultOptions;
+  protected final PipelineOptions defaultOptions;
```

Comment:
Not a fan of protected access to fields. In #2857 I simply store the options in the `TestPipeline` to keep them a smidge more decoupled. You can leave this; I'll rebase and adjust.